db/sg_replicate_cfg.go

```diff
@@ -1037,8 +1038,23 @@ func (m *sgReplicateManager) AddReplication(replication *ReplicationCfg) error {
 	return m.updateCluster(addReplicationCallback)
 }
 
+// validateReplications checks the replication configs and provides warning messages.
+func (m *sgReplicateManager) validateReplications(ctx context.Context, replications map[string]*ReplicationCfg) {
+	// the replications may exist in a way that is missing ID or name, validate and warn the user.
+	resetCheckpointMsg := "This will reset the ISGR checkpoints but prevent multiple simultaneous replications from overwriting status and checkpoint documents"
+	for replicationID, replication := range replications {
+		if replicationID == "" {
+			// we don't know the name
+			base.WarnfCtx(ctx, "An ISGR replication with an empty key exists. To fix this, this replication should be removed from database config and directly from edit the database config and the %q document. %s",
+				m.dbContext.MetadataKeys.SGCfgPrefix(m.dbContext.Options.GroupID), resetCheckpointMsg)
+		} else if replication.ID == "" {
+			base.WarnfCtx(ctx, "An ISGR replication with an empty `id` exists. To fix this, update the replication using PUT /%s/_replication/%s. %s", m.dbContext.Name, replicationID, resetCheckpointMsg)
```

Comment:
I think avoiding back-ticks in log lines would be preferred, and we decided at one point to try to standardize on ID.
```suggestion
			base.WarnfCtx(ctx, "An ISGR replication with an empty ID exists. To fix this, update the replication using PUT /%s/_replication/%s. %s", m.dbContext.Name, replicationID, resetCheckpointMsg)
```